### PR TITLE
Java doc improvements on receive API about the terminal events

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

**Review by lmolkova — APPROVED** [review]
LGTM!

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java [hunk 1]
```diff
@@ -812,6 +812,15 @@ Flux<ServiceBusReceivedMessage> peekMessages(int maxMessages, long sequenceNumbe
      *     <li>An {@link AmqpException} occurs that causes the receive link to stop.</li>
      * </ul>
      *
+     * <p>
+     * The client uses an amqp-link underneath to receive the messages; the client will transparently transition to
+     * a new amqp-link if the current one encounters a retriable error. When the client experiences a non-retriable error
```
Comment (joshfree) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java:817]: should retriable error and non-retrieable errors be links to definitions

**Review by joshfree — APPROVED** [review]

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java [hunk 2]
```diff
@@ -812,6 +812,15 @@ Flux<ServiceBusReceivedMessage> peekMessages(int maxMessages, long sequenceNumbe
      *     <li>An {@link AmqpException} occurs that causes the receive link to stop.</li>
      * </ul>
      *
+     * <p>
+     * The client uses an amqp-link underneath to receive the messages; the client will transparently transition to
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java:816]: nit: amqp-link -> "AMQP link"

"The client uses an amqp-link underneath to receive the messages;" -> "The client uses an AMQP link to receive the messages;

**Review by conniey — APPROVED** [review]
Looks good with one comment. :)

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java (hunk 1 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java]: Thanks & good point, I checked this, and there is no purely language-agnostic documentation. The one from the _Service Team_ has some generic doc but is combined with the .NET types. We should chat about developing a generic doc and then specializing it for languages.

Until there is such a Uber doc, I think adding inline examples for non-retriable errors is a good idea, as that's what the user ends up auditing. I've added examples for common cases.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java (hunk 2 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java]: thanks, updated

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java (hunk 1 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java]: The doc from _Service Team_ is [here ](https://learn.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-exceptions)

**Pull request merged**